tests/test_set_config.py

```diff
@@ -217,13 +217,8 @@ def test_incompatible_file_type(): #pylint: disable=unused-variable
     input_file = os.path.join(uwtools_file_base, pathlib.Path("fixtures/model_configure.sample"))
     args = ['-i', input_file]
 
-    outstring = io.StringIO()
-    with redirect_stdout(outstring):
+    with pytest.raises(ValueError):
         set_config.create_config_obj(args)
-    result = outstring.getvalue()
-    outcome = "Set config failure: bad file type\n"
-
-    assert result == outcome
```

Comment:
I really appreciate this magic simplification we get with pytest. This line diff is a nice illustration of what it buys us.